### PR TITLE
Support for super-fisheye lenses 

### PR DESCRIPTION
**Pull request by chhtz**
This PR adds support for super-fisheye lenses (i.e., lenses with opening angles above 180 degrees).

Essentially, the idea is to compute `theta` as `atan2(sqrt(x*x+y*y),z)` (where `[x,y,z]` is the point relative to the camera, with z pointing forward).

It is downward-compatible for points in front of the camera. Points behind the camera (i.e., negative z coordinates) gave bogus results previously and now only fail if they are exactly behind the camera.
The computation of the Jacobians should be slightly faster and more accurate, especially for z close to zero.

Other functions in the fisheye module would need to be ported as well and the documentation would also need to be updated. But before I proceed, are there any opinions on this?

### CODE REVIEW

**Comment by paroj** [comment]
without having tried it myself, maybe the omindirectional model already handles this case?
description: http://www.robots.ox.ac.uk/~cmei/articles/projection_model.pdf
implementation: http://docs.opencv.org/master/d3/ddc/group__ccalib.html#gsc.tab=0

**Comment by chhtz** [comment]
@paroj Reading only the formulae in the linked description that is an entirely different camera model -- I'm not saying that it would not be worth trying to apply this model to fisheye lenses as well, but it is not the same thing. I guess it would not really make sense to have both, otherwise.

The fact that it may make sense to join the different camera models into a common module (and try to share more code between them) is another topic ...

File: modules/calib3d/src/fisheye.cpp [hunk 1]
```diff
@@ -161,49 +160,40 @@ void cv::fisheye::projectPoints(InputArray objectPoints, OutputArray imagePoints
...
-            //Vec2d xd1 = x * cdist;
+            // double r2 = x.dot(x);
+            Vec3d dr2dom = 2 * (x[0] * dYdom[0] + x[1] * dYdom[1]);
+            Vec3d dr2dT  = 2 * (x[0] * dYdT[0]  + x[1] * dYdT[1]);
+            // double z = Y[2];
+            const Vec3d& dzdom = dYdom[2];
+            const Vec3d& dzdT  = dYdT[2];
+
+            // double theta_ = r<1e-8*z ? 1/z - r2/(3*z*z*z) : atan2(r, z)/r;
+            double dtheta_dr2 = r<1e-8*z ? -1/(3*z3) : 1/(2*r2)*(z/(r2+z2) - theta_);
```
Comment (chhtz) [inline modules/calib3d/src/fisheye.cpp]: N.B.: The small `r` derivation must actually be `+1/(6*z3)` -- I'll update this PR if I get any positive response ...

**Comment by vpisarev** [comment]
this PR will be reopened because of CI issues

**Pull request closed without merging**